src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1159,22 +1160,19 @@ static bool monitors_used_above_threshold(MonitorList* list) {
 }
 
 size_t ObjectSynchronizer::in_use_list_ceiling() {
-  // _in_use_list_ceiling is a jint so this cast could lose precision,
-  // but in reality the ceiling should never get that high.
-  return (size_t)_in_use_list_ceiling;
+  return _in_use_list_ceiling;
 }
 
 void ObjectSynchronizer::dec_in_use_list_ceiling() {
-  Atomic::add(&_in_use_list_ceiling, (jint)-AvgMonitorsPerThreadEstimate);
-#ifdef ASSERT
-  size_t l_in_use_list_ceiling = in_use_list_ceiling();
-#endif
-  assert(l_in_use_list_ceiling > 0, "in_use_list_ceiling=" SIZE_FORMAT
-         ": must be > 0", l_in_use_list_ceiling);
+  Atomic::add(&_in_use_list_ceiling, -AvgMonitorsPerThreadEstimate);
```

Comment:
Don't you need to cast AMPTE to a signed type to take the negative of it?
Why not just use Atomic::sub?

Reply:
Apparently no cast is needed since the current patch
passes Mach[1-3]. I don't remember the reason that
Atomic::add() of a negative value was used, but that's
not new in the patch so I'm planning to leave it alone.